src/Reflection/ReflectionClass.php

```diff
@@ -242,7 +242,15 @@ public static function createFromNode(
      */
     public function getShortName() : string
     {
-        return $this->node->name;
+        if (!$this->isAnonymous()) {
+            return $this->node->name;
+        }
+
+        $name = 'class@anonymous';
+        if ($this->getFileName() !== null) {
+            $name .= ' ' . $this->getFileName();
```

Comment:
Hmm, this doesn't seem consistent with PHP core, but I'm not sure (without checking `php-src` directly) what the rules are for how this should work...

Given code:

```php
<?php

$c = new class() {};

$r = new ReflectionClass($c);
var_dump($r->getShortName());
var_dump($r->isAnonymous());
```

Running from the file:

```sh
$ php test.php 
/home/james/Desktop/test.php:6:
string(58) "class@anonymous\000/home/james/Desktop/test.php0x7f89bab29019"
/home/james/Desktop/test.php:7:
bool(true)
```

Running from https://3v4l.org/vRHuk:

```
string(39) "class@anonymous/in/vRHuk0x7fb312a9a019"
bool(true)
```